packages/hub/services/validators/scheduled-payment.ts

```diff
@@ -89,23 +89,33 @@ export default class ScheduledPaymentValidator {
       }
     }
 
-    if (scheduledPayment.chainId && !isSupportedChain(scheduledPayment.chainId)) {
-      errors.chainId.push(`chain is not supported`);
-    }
+    if (scheduledPayment.chainId) {
+      if (!isSupportedChain(scheduledPayment.chainId)) {
+        errors.chainId.push(`chain is not supported`);
+      } else {
+        let feeFixedUSD =
+          this.cardpay.getConstantByNetwork('scheduledPaymentFeeFixedUSD', scheduledPayment.chainId) ?? 0;
+        if (Number(scheduledPayment.feeFixedUsd) < feeFixedUSD) {
+          errors.feeFixedUsd.push(`fee USD must be greater than or equal ${feeFixedUSD}`);
+        }
 
-    if (scheduledPayment.chainId && isSupportedChain(scheduledPayment.chainId)) {
-      let feeFixedUSD = this.cardpay.getConstantByNetwork('scheduledPaymentFeeFixedUSD', scheduledPayment.chainId) ?? 0;
-      if (Number(scheduledPayment.feeFixedUsd) < feeFixedUSD) {
-        errors.feeFixedUsd.push(`fee USD must be greater than or equal ${feeFixedUSD}`);
-      }
+        let feePercentage =
+          this.cardpay.getConstantByNetwork('scheduledPaymentFeePercentage', scheduledPayment.chainId) ?? 0;
+        let _feePercentage = Number(scheduledPayment.feePercentage);
+        if (_feePercentage < 0 || _feePercentage > 1) {
+          errors.feePercentage.push(`fee percentage must be between 0 and 1`);
+        } else if (_feePercentage < feePercentage) {
+          errors.feePercentage.push(`fee percentage must be greater than or equal ${feePercentage}`);
+        }
 
-      let feePercentage =
-        this.cardpay.getConstantByNetwork('scheduledPaymentFeePercentage', scheduledPayment.chainId) ?? 0;
-      let _feePercentage = Number(scheduledPayment.feePercentage);
-      if (_feePercentage < 0 || _feePercentage > 1) {
-        errors.feePercentage.push(`fee percentage must be between 0 and 1`);
-      } else if (_feePercentage < feePercentage) {
-        errors.feePercentage.push(`fee percentage must be greater than or equal ${feePercentage}`);
+        let supportedGasTokens = await this.cardpay.fetchSupportedGasTokens(scheduledPayment.chainId);
+        if (
+          !supportedGasTokens.find(
+            (supportedGasToken) => supportedGasToken.address === scheduledPayment.gasTokenAddress
+          )
+        ) {
+          errors.gasTokenAddress.push(`gas token is not supported`);
```

Comment:
Perhaps it would be useful to add which tokens are supported in the message

Reply:
updated